data/api/client-server/profile.yaml

```diff
@@ -222,6 +222,7 @@ paths:
                     type: string
                     description: The user's display name if they have set one, otherwise not
                       present.
+                additionalProperties: {}
```

Comment:
As I've written over at https://github.com/matrix-org/matrix-spec/pull/1909#pullrequestreview-2194861986, it's not obvious to me that we should add this. IMHO it adds no extra information.

Reply:
Continuing the discussion from https://github.com/matrix-org/matrix-spec/pull/1909#issuecomment-2247610235, I think I agree that `additionalProperties: {}` doesn't add real information when this is technically the case for almost everything in Matrix.

I have considered https://spec.matrix.org/v1.11/client-server-api/#getwell-knownmatrixclient as a counter example.

```
additionalProperties:
  type: object
  description: Application-dependent keys using Java package naming convention.
```

Using Java package naming feels like it should be the default for any custom property. However, in this case the `additionalProperties` schema also limits the type to be an object. So this feels worth keeping indeed.

Since the schema of `additionalProperties` in `/_matrix/client/v3/profile/{userId}` doesn't need any type restriction, I concur, that adding it is pointless.

However, this now makes me wonder if we should remove the remark that the response can contain additional fields when this is true for almost every response?

> This API may return keys which are not limited to `displayname` or `avatar_url`.

Reply:
> However, in this case the `additionalProperties` schema also limits the type to be an object. So this feels worth keeping indeed.

As an aside, I'm unconvinced that it is correct that other properties in the `.well-known` result must be objects, but that's a different problem.

> However, this now makes me wonder if we should remove the remark that the response can contain additional fields when this is true for almost every response?
>
>>    This API may return keys which are not limited to displayname or avatar_url.

Yes, removing that seems reasonable to me.

Reply:
> As an aside, I'm unconvinced that it is correct that other properties in the `.well-known` result must be objects, but that's a different problem.

Yeah, I was wondering about that, too. Have logged https://github.com/matrix-org/matrix-spec/issues/1913.

Reply:
> > However, this now makes me wonder if we should remove the remark that the response can contain additional fields when this is true for almost every response? [...]
>
> Yes, removing that seems reasonable to me.

Have made it so.